src/languages/javascript.coffee

```diff
@@ -607,6 +615,29 @@ JavaScriptParser.drop = (block, context, pred) ->
 
   return helper.DISCOURAGE
 
+isStandardForLoop = (node) ->
+  unless node.init? and node.test? and node.update?
+    return false
+
+  # A standard for loop starts with "var a =" or "a = "
+  if node.init.type is 'VariableDeclaration'
+    variableName = node.init.declarations[0].id.name
+  else if node.init.type is 'AssignmentExpression' and
+      node.operator is '=' and
+      node.left.type is 'Identifier'
+    variableName = node.left.name
+  else
+    return false
+
+  return (node.test.type is 'BinaryExpression' and
+      node.test.operator is '<' and
+      node.test.left.type is 'Identifier' and
+      node.test.left.name is variableName and
+      node.update.type is 'UpdateExpression' and
+      node.update.operator is '++' and
+      node.update.argument.type is 'Identifier' and
```

Comment:
Does this accept both predecrement and postdecrment?  (If so that's good.  Predecrement is arguably the better canonical form in a loop. http://stackoverflow.com/questions/561588/what-is-more-efficient-i-or-i/561609 )